starlette/staticfiles.py

```diff
@@ -156,8 +162,8 @@ async def lookup_path(
             try:
                 stat_result = await anyio.to_thread.run_sync(os.stat, full_path)
```

Comment:
os.path.realpath, os.path.commonprefix all perform blocking IO so this whole function should be in one run_sync call

Reply:
I've changed `lookup_path` to a sync method and changed it's calls to `await anyio.to_thread.run_sync(self.lookup_path, path)`